pytket/pytket/qasm/qasm.py

```diff
@@ -1039,6 +1039,45 @@ def circuit_from_qasm_wasm(
     return circuit_from_qasm(input_file, encoding=encoding, maxwidth=maxwidth)
 
 
+def circuit_from_qasm_str_wasm(
+    qasm_str: str,
+    wasm: bytes,
+    maxwidth: int = 32,
+) -> Circuit:
+    """A method to generate a tket Circuit from a qasm string and external WASM module.
+
+    :param qasm_str: wasm string
+    :param wasm: base64 encoded wasm file as a string
+    :param maxwidth: maximum allowed width of classical registers (default 32)
+    :return: pytket circuit
+    """
+    global g_parser  # noqa: PLW0602
+    wasm_module = WasmModuleHandler(wasm)
+    _set_parser(maxwidth=maxwidth)
+    assert g_parser is not None
+    cast("_CircuitTransformer", g_parser.options.transformer).wasm = wasm_module
+    return circuit_from_qasm_str(qasm_str, maxwidth=maxwidth)
+
+
+def circuit_from_qasm_str_wasmfh(
+    qasm_str: str,
+    wasmfh: WasmModuleHandler,
+    maxwidth: int = 32,
+) -> Circuit:
+    """A method to generate a tket Circuit from a qasm string and external WASM module.
+
+    :param qasm_str: wasm string
+    :param wasm: WasmModuleHandler
```

Comment:
Something descriptive would be better rather than just the type.

Reply:
done, see [879cb2e](https://github.com/CQCL/tket/pull/1898/commits/879cb2ee9fbb9c962374d5d9e741d1f1f918da59)